rispy/parser.py

```diff
@@ -120,226 +95,175 @@ def __init__(
                     continue
-            else:
-                try:
-                    yield self._parse_other(line)
-                except NextLine:
+
+                if tag == self.END_TAG:
+                    result.append(record)
+
+                    record = self._iter_till_start(lines)
                     continue
 
-    def _parse_tag(self, line):
-        tag = self.get_tag(line)
-        if tag in self.ignore:
-            raise NextLine
-
-        if tag == self.END_TAG:
-            return self.current
-
-        if tag == self.START_TAG:
-            # New entry
-            if self.in_ref:
-                raise ParseError(f"Missing end of record tag in line {self.line_number}:\n {line}")
-            self._add_tag(tag, line)
-            self.in_ref = True
-            raise NextLine
-
-        if not self.in_ref:
-            raise ParseError(f"Invalid start tag in line {self.line_number}:\n {line}")
-
-        if tag in self.mapping:
-            self._add_tag(tag, line)
-            raise NextLine
-        elif not self.skip_unknown_tags:
-            self._add_unknown_tag(tag, line)
-            raise NextLine
-
-        raise NextLine
-
-    def _parse_other(self, line):
-        if self.skip_missing_tags:
-            raise NextLine
-        if self.in_ref:
-            # Active reference
-            if self.last_tag is None:
-                raise ParseError(f"Expected tag in line {self.line_number}:\n {line}")
-            # Active tag
-            self._add_tag(self.last_tag, line, all_line=True)
-            raise NextLine
-
-        if self.is_header(line):
-            raise NextLine
-        raise ParseError(f"Expected start tag in line {self.line_number}:\n {line}")
-
-    def _add_single_value(self, name, value, is_multi=False):
+                self._add_tag(record, tag, content)
+                last_tag = tag
+
+        except StopIteration:
+            return result
+
+    def parse_line(self, line):
+        """Parse line of RIS file.
+
+        This method parses a line between the start and end tag.
+        It returns the tag and the content of the line. Typically,
+        the first 2 characters are the tag, followed by a seperator,
+        and the rest of the line is the content.
+
+        Custom parsers can override this method to change the way
+        lines are parsed. For example, a very basic RIS parser would
+        return the first 2 characters as the tag and the rest of the
+        line as the content of the tag. `(line[0:2], line[6:].strip())`
+
+        Parameters
+        ----------
+        line : str
+            Line of RIS file between start and end tag.
+
+        Returns
+        -------
+        tuple
+            Tuple containing the tag and the content of the tag.
+        """
+        if line[3:5] == " -" and line[:2].isupper():
+            return (line[0:2], line[6:].strip())
+        else:
+            return ("  ", line.strip())
```

Comment:
Maybe return `None` instead of `"  "` for the empty tag? Then the boolean value of the empty tag will be false.

Reply:
Implemented. Thanks!